src/main/java/com/petros/bibernate/util/EntityUtil.java

```diff
@@ -210,6 +203,34 @@ public static boolean isRegularField(Field field) {
         return !isEntityField(field);
     }
 
+    /**
+     * Helper method that returns entity fields that not marked with @OneToMany annotation.
+     *
+     * @param entityType type of Bibernate entity
+     * @return array of fields that are suitable for dirty checking
+     */
+    public static Field[] getEntityColumns(Class<?> entityType) {
+        return Arrays.stream(entityType.getDeclaredFields())
+                .filter(field -> !isEntityCollectionField(field))
```

Comment:
Here, we also need to exclude @ManyToOne (and @OneToOne when it becomes available).